uber/site_sections/preregistration.py

```diff
@@ -524,3 +527,22 @@ def shirt_reorder(self, session, message='', **params):
             'message': message,
             'attendee': attendee
         }
+
+    def apply_promo_code(self, session, message='', **params):
+        if 'id' and 'code' in params:
+            if params['code'] is not '':
```

Comment:
It might be better to check for `params['code'] == ''` and then set message to something like `message = 'You must enter a promo code.'`